### PR TITLE
feat(tenant-management):integrate auth0

### LINKED ISSUE #47: add auth0 support for idp configuration in tenant managemnt service
**Is your feature request related to a problem? Please describe.**

add auth0 support for idp configuration in tenant managemnt service

### PR DESCRIPTION
**Pull request by Surbhi-sharma1**
## Description

integrate auth0 in tenant management service
Fixes #47 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: services/tenant-management-service/migrations/pg/migrations/sqls/20240925102459-add-table-tenant-configs-up.sql
Comment (yeshamavani) [inline services/tenant-management-service/migrations/pg/migrations/sqls/20240925102459-add-table-tenant-configs-up.sql:1]: add the trigger to update the modified_on time whenever there is an update

File: services/tenant-management-service/src/models/dtos/tenant-dto.model.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    },
+  })
+  contacts: Contact[];
+  @property({
+    type: 'object',
+    description: 'plan details',
+    jsonSchema: getJsonSchema(Object),
+  })
+  plan: AnyObject;
+
```
Comment (yeshamavani) [inline services/tenant-management-service/src/models/dtos/tenant-dto.model.ts:35]: plans will be separate not a part of tenant

File: services/tenant-management-service/src/models/tenant-config.model.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  })
+  configKey: string;
+
+  @property({
+    type: 'object',
+    required: true,
+    name: 'config_value',
+  })
+  configValue: ConfigValue;
+
```
Comment (yeshamavani) [inline services/tenant-management-service/src/models/tenant-config.model.ts:31]: why is the configValue type defined like this 
it not specific to just auth0 and organizations 
its a generic table that can be used for anything in future
its just a json value https://github.com/sourcefuse/loopback4-microservice-catalog/blob/master/services/authentication-service/src/models/tenant-config.model.ts

File: services/tenant-management-service/src/types/i-idp.interface.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import {TenantDto} from '../models/dtos/tenant-dto.model';
+
+export enum IdPKey {
+  AUTH0 = 'auth0',
+  COGNITO = 'cognito',
+  KEYCLOAK = 'keycloak',
+}
+
+export type ConfigureIdpFunc<T> = (payload: IdpDetails) => Promise<T>;
```
Comment (yeshamavani) [inline services/tenant-management-service/src/types/i-idp.interface.ts:9]: name of this type

File: services/tenant-management-service/src/providers/idp/idp-auth0.provider.ts [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+      );
+    }
+    const configValue: ConfigValue = tenantConfig.configValue;
+    if (!configValue) {
+      throw new HttpErrors.NotFound(
+        `Tenant configuration not found for tenant: ${tenant.id}`,
+      );
+    }
+
+    const organizationData: OrganizationData = {
```
Comment (yeshamavani) [inline services/tenant-management-service/src/providers/idp/idp-auth0.provider.ts:50]: tenant_config depends on two things 
tenant_id and config_key 
please take key also into account as I can store diff types of config for a tenant

File: services/tenant-management-service/src/providers/idp/types.ts
Comment (yeshamavani) [inline services/tenant-management-service/src/providers/idp/types.ts:1]: why userdata and organizationdata types defined 
arent these defined in the auth0 management package

**Review by yeshamavani — CHANGES_REQUESTED** [review]

**Comment by yeshamavani** [comment]
Also is this https://github.com/sourcefuse/arc-saas/pull/38 PR included in this new PR ??
@Surbhi-sharma1 @Tyagi-Sunny

**Comment by Surbhi-sharma1** [comment]
https://github.com/sourcefuse/arc-saas/pull/51
closing as raised above PR

**Pull request closed without merging**